src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cleanedSourceString = sourceString.trim().replace(/;$/, '');
+
+  const result = {};
+
+  const rules = cleanedSourceString.split(';');
+
+  rules.forEach((rule) => {
+    if (rule.trim()) {
+      const [key, value] = rule.split(':').map((part) => part.trim());
```

Comment:
Consider handling cases where the `rule` does not contain a colon. Currently, `split(':')` will return an array with a single element, which could lead to unexpected behavior. You might want to check if `rule.includes(':')` before splitting.